packages/loader/container-loader/src/container.ts

```diff
@@ -664,7 +665,10 @@ export class Container
 
 		const summarizeProtocolTree =
 			this.mc.config.getBoolean("Fluid.Container.summarizeProtocolTree2") ??
-			this.loader.services.options.summarizeProtocolTree;
+			this.loader.services.options.summarizeProtocolTree ??
```

Comment:
If possible, I'd rather see different flow - Loader always supplying protocol tree. And driver dropping it if it believes it's not required / supported by this driver instance.
There could be issues with this workflow (including back-compat issues), i.e. I have not fully evaluated how feasible it is, but I'd love this flow to be considered, as it reduces leakage / need to know (by various unrelated layers) of concepts that they should probably not know about.

Reply:
In both scenarios, the change will be in loader layer. In the flow that you are mentioning, we will need to add this in every driver out there to disable this. So, I think we should enable by driver for now and not disable.

Reply:
I think long term we want that change. If anything - because we want to have simplicity in loader layer and push any new storage implementers to use single-commit. But I think it's fine to go with current approach right now.